cpython/download_pypi_top.py

```diff
@@ -106,13 +107,21 @@ def main():
     nproject = len(projs)
     print(f"Project#: {nproject}")
 
-    for index, proj in enumerate(projs, start=1):
+    def download_wrapper(args):
+        dst_dir, index, proj, nproject = args
         try:
             download_sdist(dst_dir, index, proj, nproject)
         except Exception:
             traceback.print_exc()
             cprint(f"Failed to download {proj}", "red")
 
+    max_workers = 8
```

Comment:
Can you use a value based on os.cpu_count() instead?

Reply:
Sure, but it's I/O bound, so in theory the core count doesn't change anything.

Reply:
If you are worried about memory usage, I can make it stream to disk instead of having everything in memory upfront.

Reply:
Can you add a command line argument to configure the number of threads, with a default value of 8?

Reply:
done